update-antora-spring-ui/action.yml

```diff
@@ -0,0 +1,88 @@
+name: Update Antora UI Spring
+description: A GitHub Action that detects if there is a new version of the Antora UI Spring artifact and creates a PR to update the antora-playbook.yml file to use the new version.
+
+inputs:
+  docs-branch:
+    description: The branch where the antora-playbook.yml file is. Default is docs-build
+    default: 'docs-build'
+  workflow-branch:
+    description: The branch that will be created by the workflow with the changes. Default is update-antora-ui-spring
+    default: 'update-antora-ui-spring'
+  token:
+    description: Token with write permission to pull-requests, issues and contents
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v4
+    - id: check-existing-pr
+      name: Check for Existing PR
+      run: |
+        pr_count=$(gh pr list --head ${{ inputs.workflow-branch }} --base ${{ inputs.docs-branch }} --state open --json id | jq length)
+        if [[ $pr_count -eq 0 ]]; then
+          echo "continue=true" >> $GITHUB_OUTPUT
+        fi
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.token }}
+    - name: Delete Existing Branch If No PR Open
+      if: ${{ steps.check-existing-pr.outputs.continue == 'true' }}
+      run: |
+        if git ls-remote --exit-code --heads origin "${{ inputs.workflow-branch }}" >/dev/null 2>&1; then
+          git push -d origin ${{ inputs.workflow-branch }}
+        fi
+      shell: bash
+    - name: Get Current UI Bundle URL
+      id: current
+      if: ${{ steps.check-existing-pr.outputs.continue == 'true' }}
+      run: echo current_ui_bundle_url=$(awk '/^ui:/{f=1; next} f && /^  bundle:/{f=2; next} f==2 && /^    url:/{print $2; exit}' antora-playbook.yml) >> $GITHUB_OUTPUT
```

Comment:
I think we will need to parameterize the file to search (with a default of antora-playbook.yml) because the playbook can be specified in more than way (arguments, yml, toml, etc)

For example, Spring Boot [specify the playbook file](https://github.com/spring-projects/spring-boot/blob/866e92afa698e6c96c969d3eaf2ab17cd1829404/buildSrc/src/main/java/org/springframework/boot/build/AntoraConventions.java#L193) from an [entry in package.json](https://github.com/spring-projects/spring-boot/blob/866e92afa698e6c96c969d3eaf2ab17cd1829404/antora/package.json#L16).

Reply:
I also think that the URL should be found using a regex that matches `https://github.com/spring-io/antora-ui-spring/releases/download/(latest|v\d+\.\d+\.\d+)/ui-bundle.zip` since parsing a yml file doesn't work for cases like Spring Boot

Reply:
Thanks, it is now using the following command to find the link:

```bash
grep -oE 'http[s]?://[^ ]+/ui-bundle.zip' -i -w ${{ inputs.antora-file-path }}
```